### ISSUE TITLE
Extreme Values for Temperature

### ISSUE THREAD

**Issue by juliettelavoie** [ECCC-CCCS/CMIP6-BCCAQ-Quality-Control#32]
The health checks I ran flagged extremely high daily temperature (>60 degC) in HadGEM3-GC31-LL and UKESM1-0-LL (ssp245 and ssp585). The maximum temperature hits 64 degC. The position of the maximum is not the same in both SSPs. The extremes are only present for 1 day.

![t1](https://user-images.githubusercontent.com/29236239/150848604-f870c148-5675-4b18-bf9d-c80221fe7f69.png)
![t2](https://user-images.githubusercontent.com/29236239/150848608-8eb9669e-9d15-4b05-b7bc-51c4be75a5bc.png)
![t3](https://user-images.githubusercontent.com/29236239/150848610-cf96a4d7-697d-4929-8062-fd315b35580a.png)
![t4](https://user-images.githubusercontent.com/29236239/150848612-cd6447e2-1004-4796-a747-ee65e25a68e2.png)

## How should we handle these extreme values?
Should Ouranos replace the extremely high data by a NaN before calculating their indicators? Was it removed by PCIC before calculating their indicators ? If not, removing it in the Ouranos calculations would lead to non-uniform data for all indicators...
Same question as for issue #30.

The code to reproduce the figure is in section Potential Issue 3 of the QAQC-CMIP6-BCCAQv2.ipynb.

**Comment by juliettelavoie** [comment]
After internal discussions at Ouranos, our plan is to put NaNs on pr>300 mm d-1 and tasmax>60 degC before doing the indicator calculations. Any objections?

**Comment by laura-vanvliet** [comment]
For each model and SSP I calculated the skew and kurtosis of a 10-year slice (2070-2079) for BCCAQ tasmin and tasmax. The absolute maximum over all models/SSPs is shown in figures below, and indicates that tasmin and tasmax don't have many concerning outliers.
The only thing of note, for tasmin, the large range for kurtosis comes from 4 cells (invisible at this scale) on western Vancouver Island that have large tasmin outliers (-30 C), not present on adjacent cells. These are also present in CMIP5 BCCAQ.
One caution from this might be NaNing out values based on a single threshold for all of Canada. This would miss the anomalous tasmin values. 
This analysis also doesn't recognize any abnormal outliers from high tasmax (at least for this time period).
I plan to do the same analysis for all Climdex indices to see if the abnormal values are impacting indices.
![all_mod_abs_max_sk_kurt_tasmin_tasmax](https://user-images.githubusercontent.com/77032408/151422545-8eb98946-4bc3-456c-b442-4ec276973ba3.jpg)

**Comment by juliettelavoie** [comment]
Interesting, my health checks indeed didn't flag any problem with tasmin (the threshold was -90degC).

**Comment by laura-vanvliet** [comment]
The tasmin outlier issue is not with the absolute values (only -30 C), but with their value relative to the statistical distribution of data, as well as the location. There shouldn't be temperatures this low on Vancouver Island (and aren't on adjacent cells).

**Comment by juliettelavoie** [comment]
Yes, I understand! I meant to say that I see the value in your approach, as mine didn't flag something that is clearly wrong. :)

**Comment by ssobie** [comment]
It looks like the high maximum temperature values originate from the GCMs. These values end up affecting multiple grid cells in the downscaled simulations. They do seem anomalous in the GCM simulations compared to the surrounding cells. 
The dates from the GCM may not match the downscaled dates exactly due to the reordering step. They will occur in the same month though.

![tasmax_day_HadGEM3-GC31-LL_SSP245_2094-07-15_GCM](https://user-images.githubusercontent.com/18271879/151639188-9986668d-c340-4d33-89fa-9a793282e42f.png)
![tasmax_day_HadGEM3-GC31-LL_SSP585_2064-07-19_GCM](https://user-images.githubusercontent.com/18271879/151639307-520e5c0f-2e03-409e-a2da-ae70f0a627fb.png)
![tasmax_day_UKESM1-0-LL_SSP245_2100-08-19_GCM](https://user-images.githubusercontent.com/18271879/151639315-da7d3630-b79a-430a-b42d-0b4e836e6524.png)
![tasmax_day_UKESM1-0-LL_SSP585_2097-08-05_GCM](https://user-images.githubusercontent.com/18271879/151639320-8f370bd6-8e9f-4691-8e6b-5d47b0f306c4.png)

**Comment by juliettelavoie** [comment]
Thank you @ssobie for finding the origin of the issue!

**Comment by JeremyFyke** [comment]
Yes, thanks @ssobie !

**Comment by juliettelavoie** [comment]
After more discussion, we won't do any masking as we are short on time to do it properly. As the problem is only in a few places and a few models, it should not show up in the 10,50,90 percentiles.

**Comment by ssobie** [comment]
That should be fine, the only other GCM with temperatures above 60 C in multiple cells within Canada is KACE-1-0-G. There is also one cell of CanESM5 r5i1p2f1 SSP585 on the Manitoba-US border that reaches 60.7 C, but none of the CanESM5 simulations do. None of the other GCMs exceed 60 C in Canada. 
Interestingly, the UKESM1-0-LL GCM simulates temperatures up to 86.5 C in the US and has a global maximum of 98 C on the west coast of Australia.

**Comment by JeremyFyke** [comment]
Hi all: we have heard back from UKESM/HadGEM3 folks re: this issue.  These spikes are a known/documented problem, see model errata here:

https://errata.es-doc.org/static/view.html?uid=76b3f818-d65f-c76b-bfd8-cae5bc27825c

In short, they assess issue to be sporadic inability of atmospheric subgrid mixing model component to mix away accumulated heat in a surface grid cell by sensible heat transfer, in a particular model timestep.  This is a problem that would only impact maximum temperatures, not minimum temperatures.
